### ISSUE TITLE
`the role was not found` error in offline mode

### ISSUE THREAD

**Issue by Aohzan** [ansible/ansible-lint#3174]
##### Summary

Hello,

Since version 6.12.0, I can't run ansible-lint on my playbooks in offline mode (used in CI), it raises this checks for the first external role called in any playbook:

```
syntax-check[specific]: the role 'xxx' was not found in /src/plays/roles:/root/.cache/ansible-compat/25a663/roles:/src/roles:/src/roles:/src/plays
plays/xxx.yml:12:15
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 6.14.1 using ansible 2.14.3
```

In a docker image build `FROM debian:bullseye-slim`

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Install ansible-lint from version 6.12.0 to 6.14.2, without having installed requirements.yml, run

```console
ansible-lint --offline
```

##### Desired Behavior

Don't try to resolve roles which doesn't exist in local directories

##### Actual Behavior

Raise the error

##### Others informations

syntax-check failed logically, but in ansible-lint <6.12.0, it was not a problem

```console
ansible-playbook --syntax-check plays/bdd.yml 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
ERROR! the role 'xxxx' was not found in /src/plays/roles:/src/roles:/src/plays

The error appears to be in '/src/plays/xxx.yml': line 17, column 15, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      ansible.builtin.import_role:
        name: xxx
              ^ here
```

playbook:

```yaml
- name: playbook
  hosts: my_hosts
  tasks:
    - name: Import role xxx
      ansible.builtin.import_role:
        name: xxx
```

requirements:

```yaml
---
roles:
  - name: xxxx
    src: git@xxx/xxxx.git
    scm: git
    version: "1.0.0"
```

**Comment by ssbarnea** [comment]
If you want to pass the linting in offline mode, it is up to you to install the role before running the linting or at least to use the mocking feature to mock missing roles.

We will not implement a feature to ignore any unknown role call as that would neuter one of the most useful linter features.
